google/cloud/bigtable/admin/internal/wait_for_consistency.cc

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/admin/internal/wait_for_consistency.h"
+#include "google/cloud/bigtable/admin/bigtable_table_admin_client.h"
+#include "google/cloud/bigtable/admin/bigtable_table_admin_options.h"
+#include "google/cloud/bigtable/admin/internal/bigtable_table_admin_option_defaults.h"
+#include <chrono>
+
+namespace google {
+namespace cloud {
+namespace bigtable_admin_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using RespType = StatusOr<bigtable::admin::v2::CheckConsistencyResponse>;
+using TimerFuture = future<StatusOr<std::chrono::system_clock::time_point>>;
+
+// This class borrows heavily from `google::cloud::internal::AsyncRetryLoop`
+class AsyncWaitForConsistencyImpl
+    : public std::enable_shared_from_this<AsyncWaitForConsistencyImpl> {
+ public:
+  AsyncWaitForConsistencyImpl(CompletionQueue cq,
+                              bigtable_admin::BigtableTableAdminClient client,
```

Comment:
Should this parameter  `std::shared_ptr<bigtable_admin::BigtableTableAdminConnection>` ?  Otherwise you will have the ownership cycle again?